chemprop/cli/train.py

```diff
@@ -542,170 +531,220 @@ def main(args):
+                predictor_cls,
+                input_dim=mp_block.output_dim + d_xf,
+                n_tasks=train_dset.datasets[0].Y.shape[1]
+                if multicomponent
+                else train_dset.Y.shape[1],
+                hidden_dim=args.ffn_hidden_dim,
+                n_layers=args.ffn_num_layers,
+                dropout=args.dropout,
+                activation=args.activation,
+                criterion=criterion,
+                n_classes=args.multiclass_num_classes,
+                spectral_activation=args.spectral_activation,
+            )
+
+            if isinstance(predictor_ffn, RegressionFFN):
+                scaler = train_dset.normalize_targets()
+                val_dset.normalize_targets(scaler)
+                logger.info(f"Train data: loc = {scaler.mean_}, scale = {scaler.scale_}")
+            else:
+                scaler = None
+
+            train_loader = data.MolGraphDataLoader(train_dset, args.batch_size, args.num_workers)
+            val_loader = data.MolGraphDataLoader(
+                val_dset, args.batch_size, args.num_workers, shuffle=False
+            )
+
+            if multicomponent:
+                if len(test_data[0]) > 0:
+                    test_dsets = [make_dataset(data, args.rxn_mode) for data in test_data[fold_idx]]
+                    test_dset = data.MulticomponentDataset(test_dsets)
+                    test_loader = data.MolGraphDataLoader(
+                        test_dset, args.batch_size, args.num_workers, shuffle=False
+                    )
+                else:
+                    test_loader = None
+            else:
+                if len(test_data) > 0:
+                    test_dset = make_dataset(test_data[fold_idx], args.rxn_mode)
+                    test_loader = data.MolGraphDataLoader(
+                        test_dset, args.batch_size, args.num_workers, shuffle=False
+                    )
+                else:
+                    test_loader = None
+
+            mpnn_cls = MulticomponentMPNN if multicomponent else MPNN
+            model = mpnn_cls(
+                mp_block,
+                agg,
+                predictor_ffn,
+                True,
+                None,
+                args.task_weights,
+                args.warmup_epochs,
+                args.init_lr,
+                args.max_lr,
+                args.final_lr,
+            )
+            logger.info(model)
+
+            monitor_mode = "min" if model.metrics[0].minimize else "max"
+            logger.debug(f"Evaluation metric: '{model.metrics[0].alias}', mode: '{monitor_mode}'")
+
+            if no_cv:
+                output_dir = args.output_dir
+            else:
+                output_dir = args.output_dir / f"fold_{fold_idx}"
+
+            tb_logger = TensorBoardLogger(output_dir / "tb_logs")
+            checkpointing = ModelCheckpoint(
+                output_dir / "chkpts",
+                "{epoch}-{val_loss:.2f}",
```

Comment:
Is it possible for this to overwrite previous checkpoints?

Reply:
yes, provided they have the `output_dir` and same validation loss (to 2 decimal points) on the same epoch

Reply:
With changes implemented in https://github.com/chemprop/chemprop/pull/599, I think `output_dir` would be `$PWD/chemprop_training/<input_stem>/$NOW` so this will not be overwritten.